templates-java/src/test/java/br/com/templates_java/config/jms_aws_sqs/ProducerMessageGeneratorTest.java

```diff
@@ -0,0 +1,65 @@
+package br.com.templates_java.config.jms_aws_sqs;
+
+import br.com.generator.core.GeneratorOptions;
+import br.com.templates_java.helper.LoadTemplateTester;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+public class ProducerMessageGeneratorTest {
+
+    private LoadTemplateTester loadTemplateTester;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private File temporaryPath;
+
+    @Before
+    public void setUp() throws IOException {
+        temporaryPath = temporaryFolder.newFolder("test-templates");
+        loadTemplateTester = new LoadTemplateTester();
+    }
+
+    @Test
+    public void shouldCreateFile() throws IOException {
+        GeneratorOptions generatorOptions = new GeneratorOptions();
+        generatorOptions.setDestination(temporaryPath.getAbsolutePath());
+
+        HashMap<String, String> keyValue = new HashMap<String, String>();
+        keyValue.put("${package}", "br.com.example");
```

Comment:
Be cautious when using user-provided input (`${package}`) to avoid potential security vulnerabilities like path traversal attacks.